jfr-daemon/src/main/java/com/newrelic/jfr/daemon/JFRController.java

```diff
@@ -142,7 +142,13 @@ private static DaemonConfig buildConfig() {
 
   static JFRUploader buildUploader(DaemonConfig config)
       throws UnknownHostException, MalformedURLException {
-    String localIpAddr = InetAddress.getLocalHost().toString();
+    String localIpAddr;
+    try {
+      localIpAddr = InetAddress.getLocalHost().toString();
+    } catch (Throwable e) {
+      logger.error("Unable to get localhost IP, defaulting to 127.0.0.1.", e);
+      localIpAddr = "127.0.0.1";
```

Comment:
```suggestion
      localIpAddr = InetAddress.getLoopbackAddress();
```

This is what we do in the agent.

Reply:
Thanks, that looks more legitimate.